src/rdkafka_cgrp.c

```diff
@@ -3886,9 +3886,8 @@ static void rd_kafka_cgrp_revoke_all_rejoin (rd_kafka_cgrp_t *rkcg,
                            "current assignment and rebalance");
         }
 
-
-        if (rkcg->rkcg_group_assignment) {
-
+        if (rkcg->rkcg_group_assignment &&
```

Comment:
So if a consumer joins a group and gets an empty assignment, it will never get a revoke rebalance callback?
I'm wondering if this change in signalling poses a problem for applications that needs to track whether it has an assignment or not, whether the group rebalances or not.

I think it'd be good to update the rdkafka.h docs to make it clear that there will be no empty revoke callbacks (even if there were empty assignments) in the cooperative case.

Reply:
> So if a consumer joins a group and gets an empty assignment, it will never get a revoke rebalance callback?

correct. this is outlined in the kip, and mirrored in the java docs (https://kafka.apache.org/26/javadoc/index.html?org/apache/kafka/clients/consumer/KafkaConsumer.html):

> With cooperative rebalancing, onPartitionsRevoked(Collection) and onPartitionsLost(Collection) will only be triggered when there are non-empty partitions revoked or lost from this consumer member during a rebalance event.

on a related note: in java, for the EAGER case, revoke/assign symmetry is not retained in the case of lost partitions. In librdkafka, we do retain this symmetry. But we follow what the Java consumer does exactly for the COOPERATIVE case.

will update rdkafka.h